### PR TITLE
Add form logic for pillar analysis

### PR DESCRIPTION
**Pull request by tnagorra**
Depends https://github.com/toggle-corp/re-map/pull/12
Depends https://github.com/toggle-corp/re-map/pull/13

### CODE REVIEW

File: src/redux/selectors/route.js [hunk 1]
```diff
@@ -56,6 +56,5 @@ export const entryIdFromRouteSelector = createRouteSelector('entryId');
 export const leadGroupIdFromRouteSelector = createRouteSelector('leadGroupId');
 export const projectIdFromRouteSelector = createRouteSelector('projectId');
 export const userIdFromRouteSelector = createRouteSelector('userId');
-export const questionnaireIdFromRouteSelector = createRouteSelector('questionnaireId');
-export const analysisIdFromRouteSelector = createRouteSelector('analysisId');
+export const questionnaireIdFromRouteSelector = createRouteSelector('questionnaireId'); export const analysisIdFromRouteSelector = createRouteSelector('analysisId');
```
Comment (AdityaKhatri) [inline src/redux/selectors/route.js]: Fix plz

File: src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    PartialForm,
+    Error,
+} from '@togglecorp/toggle-form';
+
+import { AnalyticalEntryType } from '../../schema';
+
+import styles from './styles.scss';
+
+interface AnalyticalEntryInputProps {
+   value: PartialForm<AnalyticalEntryType>,
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx]: ; is missing

File: src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+   error: Error<AnalyticalEntryType> | undefined;
+   // onChange: (value: PartialForm<AnalyticalEntryType>, index: number) => void;
+   onRemove: (index: number) => void;
+   index: number,
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx]: ; is missing

File: src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        index,
+    } = props;
+
+    // const onFieldChange = useFormObject(index, value, onChange);
+
+    return (
+        <div className={styles.entry}>
+            {error?.$internal && (
+                <p>
+                    {error?.$internal}
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx]: maybe error.$internal

File: src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                </p>
+            )}
+            <h4>
+                {value.entry}
+            </h4>
+            <QuickActionButton
+                name={index}
+                onClick={onRemove}
+                title="Remove Analytical Entry"
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx:46]: we can use string translations

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                        </QuickActionButton>
+                        <QuickActionButton>
+                            <BiBarChartSquare />
+                        </QuickActionButton>
+                    </div>
+                    <QuickActionButton
+                        className={styles.button}
+                        name={index}
+                        onClick={onRemove}
+                        title="Remove Analytical Statement"
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx:88]: I think we can use string translations

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                    </QuickActionButton>
+                    <QuickActionButton
+                        className={styles.button}
+                    >
+                        <GrDrag />
+                    </QuickActionButton>
+                </header>
+                {error?.$internal && (
+                    <p className={styles.error}>
+                        {error?.$internal}
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx]: can't we just use error.$internal

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    } = useFormArray('analyticalEntries', value.analyticalEntries ?? [], onFieldChange);
+
+    const handleAnalyticalEntryAdd = useCallback(
+        () => {
+            const uuid = randomString();
+            const newAnalyticalEntry: PartialForm<AnalyticalEntryType> = {
+                uuid,
+                // FIXME: add order
+                order: 0,
+                entry: Math.ceil(Math.random() * 100),
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx:61]: why are we using this random number for? Isn't uuid sufficient?

File: src/views/PillarAnalysis/index.tsx [hunk 9]
```diff
@@ -298,13 +407,32 @@ function PillarAnalysis(props: PageProps) {
...
+                                value={analyticalStatement}
+                                onChange={onAnalyticalStatementChange}
+                                onRemove={onAnalyticalStatementRemove}
+                                // eslint-disable-next-line max-len
+                                error={error?.fields?.analyticalStatements?.members?.[analyticalStatement.uuid]}
+                            />
+                        ))}
+                        <QuickActionButton
+                            className={styles.addStatementButton}
+                            name={undefined}
```
Comment (samshara) [inline src/views/PillarAnalysis/index.tsx:439]: why are we using undefined here?

File: src/views/PillarAnalysis/index.tsx [hunk 10]
```diff
@@ -298,13 +407,32 @@ function PillarAnalysis(props: PageProps) {
...
+                            onClick={handleAnalyticalStatementAdd}
+                            title="Add Analytical Statement"
```
Comment (samshara) [inline src/views/PillarAnalysis/index.tsx:442]: we should use string translations here

File: src/views/PillarAnalysis/index.tsx [hunk 11]
```diff
@@ -214,10 +312,14 @@ function PillarAnalysis(props: PageProps) {
...
                 )}
             >
-                {breadcrumb(pillarAnalysis?.analysisName, pillarAnalysis?.title ?? '')}
+                {breadcrumb(pillarAnalysis?.analysisTitle, pillarAnalysis?.title ?? '')}
             </FullPageHeader>
             <div className={styles.content}>
-                {pending && <LoadingAnimation />}
+                {error?.$internal && (
+                    <p>
+                        {error?.$internal}
```
Comment (samshara) [inline src/views/PillarAnalysis/index.tsx]: maybe error.$internal

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx (hunk 8 above)
Comment (tnagorra) [reply src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx:61]: This is temporary. Will remove this after we can drag/drop the entry from sidepane

File: src/views/PillarAnalysis/index.tsx (hunk 9 above)
Comment (tnagorra) [reply src/views/PillarAnalysis/index.tsx:439]: This is a typing issue!

**Pull request merged**